### PR TITLE
chore(cik8s,doks): switch to official Datadog Docker image without custom checks

### PR DESCRIPTION
**Pull request by lemeurherve**
As we recently encountered a lot of ["Weird response time" alerts](https://github.com/jenkins-infra/datadog/blob/bd70e30624fe58d4b1a75d045468febb0da3e0a7/datadog-monitors.tf#L93-L121) from `cik8s` cluster (probably due to a DNS problem), this PR switches to official Datadog Docker image for all clusters by default, and use [our own Datadog image with custom checks](https://github.com/jenkins-infra/docker-datadog/blob/main/conf.d/http_check.d/jenkins.yaml) on public facing clusters.

### CODE REVIEW

File: config/datadog_cik8s.yaml [hunk 1]
```diff
@@ -11,3 +11,9 @@ agents:
       operator: "Equal"
       value: "true"
       effect: "NoSchedule"
+  image:
+    # official image without custom checks
+    repository: "datadog/agent"
+    tag: "7"
+    # we don't have to skip version test as we're not using the docker image digest
+    doNotCheckTag: false
```
Comment (dduportal) [inline config/datadog_cik8s.yaml:19]: Is there a reason to not pick the default values from the chart, aka. https://github.com/DataDog/helm-charts/blob/93c4fe5f2e58537a25626bee9656a21ef30415d8/charts/datadog/values.yaml#L1191-L1212 ?

**Review by dduportal — DISMISSED** [review]
LGTM under the assumption that the override here is temporary and should be inverted (e.g. in long term, we should not change images unless we want the custom image)

**Comment by lemeurherve** [comment]
I've inversed the logic: by default official Datadog image, custom image on public clusters only

File: config/datadog_doks-public.yaml [hunk 2]
```diff
@@ -4,3 +4,10 @@ clusterAgent:
   image:
     pullSecrets:
       - name: "dockerhub-credential"
+agent:
```
Comment (dduportal) [inline config/datadog_doks-public.yaml:7]:
```suggestion
agents:
```

File: config/datadog_eks-public.yaml [hunk 3]
```diff
@@ -4,3 +4,10 @@ clusterAgent:
   image:
     pullSecrets:
       - name: "dockerhub-credential"
+agent:
```
Comment (dduportal) [inline config/datadog_eks-public.yaml:7]:
```suggestion
agents:
```

File: config/datadog_prodpublick8s.yaml [hunk 4]
```diff
@@ -8,3 +8,10 @@ datadog:
     hostCAPath: /etc/kubernetes/certs/kubeletserver.crt
     # Required as of Agent 7.35 because Kubelet certificates in AKS do not have a Subject Alternative Name (SAN) set.
     tlsVerify: false
+agent:
```
Comment (dduportal) [inline config/datadog_prodpublick8s.yaml:11]:
```suggestion
agents:
```

File: config/datadog_publick8s.yaml [hunk 5]
```diff
@@ -8,3 +8,10 @@ datadog:
     hostCAPath: /etc/kubernetes/certs/kubeletserver.crt
     # Required as of Agent 7.35 because Kubelet certificates in AKS do not have a Subject Alternative Name (SAN) set.
     tlsVerify: false
+agent:
```
Comment (dduportal) [inline config/datadog_publick8s.yaml:11]:
```suggestion
agents:
```

**Review by dduportal — CHANGES_REQUESTED** [review]
Almost there! typo ;)

**Pull request merged**